### ISSUE TITLE
[BUG]: Item currency gets deleted if server not shut down properly

### ISSUE THREAD

**Issue by r0xANDt0l** [TheNewEconomy/TNE-Bukkit#522]
### TNE Version

0.1.1.17-PRE3

### Spigot Version

1.19.3

### Your config.yml

https://paste.ee/p/iy8Ie

### Your latest server log

https://mclo.gs/ksM7kkY

### What part of TNE is the bug affecting?

Other Plugin Support

### Is your server using online mode?

Yes

### Bug description

If the server doesn't shut down properly (for example, if it crashes), all items that are used as currency in the users who were online in the crash get deleted

### How to reproduce

1. Have people online in a server with an item currency
2. Server crash
3. When it restarts, those people don't have the currency in their inventory

**Comment by creatorfromhell** [comment]
Do they get deleted, or does it not save their balances? This could be an issue resolved by having your server occasionally run /tne save.

**Comment by r0xANDt0l** [comment]
latest server log doesn't have a crash

**Comment by r0xANDt0l** [comment]
> Do they get deleted, or does it not save their balances? This could be an issue resolved by having your server occasionally run /tne save.

The items were gone from their inventories, and if they did /bal, they couldn't check it. I usually did `/tne save` but since you never know when a crash is going to happen, I can't do it. I'll try to do the /tne save every minute or so using a pterodactyl task

**Comment by creatorfromhell** [comment]
> > Do they get deleted, or does it not save their balances? This could be an issue resolved by having your server occasionally run /tne save.
> 
> The items were gone from their inventories, and if they did /bal, they couldn't check it. I usually did `/tne save` but since you never know when a crash is going to happen, I can't do it. I'll try to do the /tne save every minute or so using a pterodactyl task

Yeah, crashes would stop any fix from being implemented. I can't include code to save balances any time an item is picked up or dropped as that would cause insufferable TPS drops.

**Comment by r0xANDt0l** [comment]
Maybe for items, it could just read the amount of X item set as currency in the inventory, and do the calculations from there?

**Comment by creatorfromhell** [comment]
> Maybe for items, it could just read the amount of X item set as currency in the inventory, and do the calculations from there?

The issue with that is any offline transactions would cause a dupe race-condition that wouldn't be worth it just because a crash that should be resolved isn't.

**Comment by r0xANDt0l** [comment]
> The issue with that is any offline transactions would cause a dupe race-condition that wouldn't be worth it just because a crash that should be resolved isn't.
True
